src/tir/schedule/primitive/cache_read_write.cc

```diff
@@ -1297,6 +1603,317 @@ StmtSRef CacheWrite(ScheduleState self, const StmtSRef& block_sref, int write_bu
   return result_block_sref;
 }
 
+/*! \brief A visitor that collects variables appeared in expressions, stored in `touched` field.*/
+class VarCollector : public ExprVisitor {
```

Comment:
Is it the same as https://github.com/yzh119/tvm/blob/7bca3c2ca9fb144244f1a6b5968f5911d47a956b/include/tvm/tir/analysis.h#L102

Reply:
Thanks, I'll use `UndefinedVars` instead.

Reply:
`UndefinedVars` only accepts `Stmt`/`Expr` as inputs while `BufferRegion` does not belong to any of them. Actually, we can use it in our case by calling `UndefinedVars` on each of the `range->min` and merge those results but apparently, it's not elegant. `UndefinedVars` depends on `VarUseDefAnalaysis`, and using this visitor would be much more convenient.

`VarUseDefAnalysis` resides in `split_host_device.cc` which is not exposed in headers. I wonder would you mind me refactoring the code to expose it in some header like `analysis.h`?